src/voxelisers/bvh-ray-thick.ts

```diff
@@ -0,0 +1,140 @@
+        LOG('BVH created...');
+
+        // Generate rays
+        const bounds = mesh.getBounds();
+        bounds.min.floor();
+        bounds.max.ceil();
+
+        const planeDims = Vector3.sub(bounds.max, bounds.min).add(1);
+        const numRays = (planeDims.x * planeDims.y) + (planeDims.x * planeDims.z) + (planeDims.y * planeDims.z);
+        const rays = new Array<Ray>(numRays);
+        let rayIndex = 0;
+        {
+            // Generate x-plane rays
+            for (let y = bounds.min.y; y <= bounds.max.y; ++y) {
+                for (let z = bounds.min.z; z <= bounds.max.z; ++z) {
+                    rays[rayIndex++] = {
+                        origin: new Vector3(bounds.min.x - 1, y, z),
+                        axis: Axes.x,
+                    };
+                }
+            }
+            // Generate y-plane rays
+            for (let x = bounds.min.x; x <= bounds.max.x; ++x) {
+                for (let z = bounds.min.z; z <= bounds.max.z; ++z) {
+                    rays[rayIndex++] = {
+                        origin: new Vector3(x, bounds.min.y - 1, z),
+                        axis: Axes.y,
+                    };
+                }
+            }
+            // Generate z-plane rays
+            for (let x = bounds.min.x; x <= bounds.max.x; ++x) {
+                for (let y = bounds.min.y; y <= bounds.max.y; ++y) {
+                    rays[rayIndex++] = {
+                        origin: new Vector3(x, y, bounds.min.z - 1),
+                        axis: Axes.z,
+                    };
+                }
+            }
+        }
+        ASSERT(rays.length === rayIndex);
+        LOG('Rays created...');
+
+        const VecZero = new Vector3(0, 0, 0)
+
+        // Ray test BVH
+        const taskHandle = ProgressManager.Get.start('Voxelising');
+        for (rayIndex = 0; rayIndex < rays.length; ++rayIndex) {
+            ProgressManager.Get.progress(taskHandle, rayIndex / rays.length);
+
+            const ray = rays[rayIndex];
+            const intersections = bvh.intersectRay(ray.origin, axesToDirection(ray.axis), false);
+            for (const intersection of intersections) {
+                const point = intersection.intersectionPoint;
+                const position = new Vector3(point.x, point.y, point.z);
+                
+                // // Shrinking towards the center
+                // const centerVector = VecZero.copy().sub(position)
+                // const depthPosition = position.copy().add(centerVector.normalise()).round()
+                
+                // Shrinking towards the perpendicular vector
+                const triangle = intersection.triangle
+                const p0 = new Vector3(triangle[0].x, triangle[0].y, triangle[0].z)
+                const p1 = new Vector3(triangle[1].x, triangle[1].y, triangle[1].z)
+                const p2 = new Vector3(triangle[2].x, triangle[2].y, triangle[2].z)
+                const v0 = p1.sub(p0)
```

Comment:
This should be `const v0 = Vector3.sub(p1, p0);` as currently this modifies `p1`.